tests/e2e/csi_static_provisioning_file_basic.go

```diff
@@ -0,0 +1,239 @@
+			}
+		}
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Datastore is not found in the datacenter list")
+	})
+
+	/*
+		This test verifies the static provisioning workflow.
+
+		Test Steps:
+		1. Create File share and get the file share id.
+		2. Create PV Spec with volumeID set to file share ID created in Step-1, and PersistentVolumeReclaimPolicy is set to Delete.
+		3. Create PVC with the storage request set to PV's storage capacity.
+		4. Wait for PV and PVC to bound.
+		5. Create a POD.
+		6. Verify volume is attached to the node and volume is accessible in the pod by creating a file inside volume.
+		7. Verify container volume metadata is present in CNS cache.
+		8. Delete POD.
+		9. Delete PVC.
+		10. Verify PV is deleted automatically.
+	*/
+	ginkgo.It("Verify basic static provisioning workflow", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		err := connectCns(ctx, &e2eVSphere)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Creating file share")
+		cnsCreateReq := cnstypes.CnsCreateVolume{
+			This:        cnsVolumeManagerInstance,
+			CreateSpecs: []cnstypes.CnsVolumeCreateSpec{*getFileShareCreateSpec(defaultDatastore.Reference())},
+		}
+		cnsCreateRes, err := cnsmethods.CnsCreateVolume(ctx, e2eVSphere.CnsClient.Client, &cnsCreateReq)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		task, err := object.NewTask(e2eVSphere.Client.Client, cnsCreateRes.Returnval), nil
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		taskInfo, err := cns.GetTaskInfo(ctx, task)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		fileShareVolumeID := taskResult.GetCnsVolumeOperationResult().VolumeId.Id
+
+		// Deleting the volume with deleteDisk set to false
+		ginkgo.By("Deleting the fileshare with deleteDisk set to false")
+		cnsDeleteReq := cnstypes.CnsDeleteVolume{
+			This:       cnsVolumeManagerInstance,
+			VolumeIds:  []cnstypes.CnsVolumeId{{Id: fileShareVolumeID}},
+			DeleteDisk: false,
+		}
+		cnsDeleteRes, err := cnsmethods.CnsDeleteVolume(ctx, e2eVSphere.CnsClient.Client, &cnsDeleteReq)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		task, err = object.NewTask(e2eVSphere.Client.Client, cnsDeleteRes.Returnval), nil
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		taskInfo, err = cns.GetTaskInfo(ctx, task)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		_, err = cns.GetTaskResult(ctx, taskInfo)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		staticPVLabels := make(map[string]string)
+		staticPVLabels["fileshare-id"] = strings.TrimPrefix(fileShareVolumeID, "file:")
+
+		ginkgo.By("Creating the PV")
+		pv = getPersistentVolumeSpecForFileShare(fileShareVolumeID, v1.PersistentVolumeReclaimDelete, staticPVLabels, v1.ReadOnlyMany)
+		pv, err = client.CoreV1().PersistentVolumes().Create(ctx, pv, metav1.CreateOptions{})
+		if err != nil {
+			return
```

Comment:
Shouldnt this be `gomega.Expect(err).NotTo(gomega.HaveOccurred())`

Reply:
Done